v4/examples/langchain_langsmith.mdx

```diff
@@ -0,0 +1,428 @@
+
+- Username
+- API key
+- Update the dateFrom and to dateTo variables
+  ‍
+
+```curl
+curl --location 'https://api.cal.com/v1/availability?apiKey=cal_live_xxxxxxxxxxxxxx&dateFrom=2024-04-15T00%3A00%3A00.000Z&dateTo=2024-04-22T00%3A00%3A00.000Z&username=michael-louis-xxxx'
+```
+
+You should get a response similar to the following:
+
+```
+{
+    "busy": [
+        {
+            "start": "2024-04-15T13:00:00.000Z",
+            "end": "2024-04-15T13:30:00.000Z"
+        },
+        {
+            "start": "2024-04-22T13:00:00.000Z",
+            "end": "2024-04-22T13:30:00.000Z"
+        },
+        {
+            "start": "2024-04-29T13:00:00.000Z",
+            "end": "2024-04-29T13:30:00.000Z"
+        },
+	   ....
+    ],
+    "timeZone": "America/New_York",
+    "dateRanges": [
+        {
+            "start": "2024-04-15T13:45:00.000Z",
+            "end": "2024-04-15T16:00:00.000Z"
+        },
+        {
+            "start": "2024-04-15T16:45:00.000Z",
+            "end": "2024-04-15T19:45:00.000Z"
+        },
+	    ....
+        {
+            "start": "2024-04-19T18:45:00.000Z",
+            "end": "2024-04-19T21:00:00.000Z"
+        }
+    ],
+    "oooExcludedDateRanges": [
+
+    ],
+    "workingHours": [
+        {
+            "days": [
+                1,
+                2,
+                3,
+                4,
+                5
+            ],
+            "startTime": 780,
+            "endTime": 1260,
+            "userId": xxxx
+        }
+    ],
+    "dateOverrides": [],
+    "currentSeats": null,
+    "datesOutOfOffice": {}
+}
+```
+
+Great! Now we know that our API key is working and pulling information from our calendar. The API calls we will be using later in this tutorial are:
+
+- **/availability**: Get your availability
+- **/bookings**: Book a slot
+
+### Cerebrium setup
+
+If you don’t have a Cerebrium account, you can create one by signing up [here](https://dashboard.cerebrium.ai/register) and following the documentation [here](https://docs.cerebrium.ai/cerebrium/getting-started/installation) to get setup
+
+In your IDE, run the following command to create our Cerebrium starter project: cerebrium init agent-tool-calling. This creates two files:
+
+- Main.py - Our entrypoint file where our code lives
+- cerebrium.toml - A configuration file that contains all our build and environment settings
+  ‍
+
+Add the following pip packages near the bottom of your cerebrium.toml. This will be used in creating our deployment environment.
+
+```
+[cerebrium.dependencies.pip]
+pydantic = "latest"
+langchain = "latest"
+pytz = "latest" ##this is used for timezones
+openai = "latest"
+langchain_openai = "latest"
+```
+
+We will be using OpenAI GPT3.5 for our use cases and so we need an API key from them. If you don’t have an account, you can sign up [here](https://openai.com/). You can then create an API key [here](https://platform.openai.com/settings/profile?tab=api-keys). The API key should be in the format: “sk_xxxxx”.
```

Comment:
User keys have been deprecated in favour of project keys. May want to update that
```suggestion
We will be using OpenAI GPT3.5 for our use cases and so we need an API key from them. If you don’t have an account, you can sign up [here](https://openai.com/). You can then create an API key [here](https://platform.openai.com/api-keys). The API key should be in the format: “sk_xxxxx”.
```